controllers/usernamespace/usernamespace_controller.go

```diff
@@ -561,7 +521,7 @@ func (r *CheUserNamespaceReconciler) reconcileGitTlsCertificate(ctx context.Cont
 
 func (r *CheUserNamespaceReconciler) reconcileNodeSelectorAndTolerations(ctx context.Context, targetNs string, checluster *chev2.CheCluster, deployContext *chetypes.DeployContext) error {
 	ns := &corev1.Namespace{}
-	if err := r.client.Get(ctx, client.ObjectKey{Name: targetNs}, ns); err != nil {
+	if err := r.nonCachedClient.Get(ctx, client.ObjectKey{Name: targetNs}, ns); err != nil {
```

Comment:
Is there a reason for switching to the non-cached client? For `Get` requests, this will lead to a lot more API traffic.

Reply:
I see, I will revert my changes.
The idea behind those changes was to slightly reduce memory consumption since we don't often update namespace to set node selector and toleration.

Reply:
If it's very infrequent, then non-cached would be fine. The main difference is that with the cached client, any read requests (e.g. get, list, etc.) are free since it only ever checks the internal cache -- for the non-cached client, every time we get or list in a reconcile it has to make a k8s API request, which is slower and also hits the API server. If this reconcile runs infrequently, it should be fine to choose either. "Infrequent" here means reconciles, however, rather than updates to namespaces. 

We might also be able to limit memory usage by restricting which namespaces are cached or doing something similar.